### PR TITLE
added detection & dispatching of some modern NEON instructions (NEON_FP16, NEON_BF16)

### PR DESCRIPTION
**Pull request by vpisarev**
Currently, platform-specific (NEON) code is required to make use of those instructions. Later on, maybe universal intrinsics for FP16 and BF16 arithmetics will be added. Note that even modern ARM platforms don't have full set of BF16 operations. This is mostly instructions to implement BF16xBF16 to FP32 matrix multiplication.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asmorkalov** [comment]
Looks like the mentioned features are supported in `/proc/self/auxv`. No need to rework the approach.
See profiles from ARM: https://developer.arm.com/downloads/-/exploration-tools/feature-names-for-a-profile and constants for HWCAP: https://www.kernel.org/doc/html/v6.1/arm64/elf_hwcaps.html

**Comment by asmorkalov** [comment]
```
The SIGILL and SIGTERM signals aren't generated under Windows. They're included for ANSI compatibility. Therefore, you can set signal handlers for these signals by using signal, and you can also explicitly generate these signals by calling [raise](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/raise?view=msvc-170).
```
Link: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/signal?view=msvc-170

**Comment by opencv-alalek** [comment]
> `/proc/self/auxv`

There is a header of actual auxv flags / values: https://elixir.bootlin.com/linux/latest/source/arch/arm64/include/uapi/asm/hwcap.h#L106

Also some libraries parse `/proc/cpuinfo` (but auxv is preferable)

File: cmake/OpenCVCompilerOptimizations.cmake [hunk 1]
```diff
@@ -343,12 +345,17 @@ elseif(ARM OR AARCH64)
     endif()
     ocv_update(CPU_FP16_IMPLIES "NEON")
   else()
-    ocv_update(CPU_KNOWN_OPTIMIZATIONS "NEON;FP16;NEON_DOTPROD")
+    ocv_update(CPU_KNOWN_OPTIMIZATIONS "NEON;FP16;NEON_DOTPROD;FP16_SIMD;BF16_SIMD")
```
Comment (opencv-alalek) [inline cmake/OpenCVCompilerOptimizations.cmake:348]: > FP16_SIMD;BF16_SIMD

How does scope of these instructions correlate with other platforms?
... and available universal intrinsics?

Until it is unclear, it is better to add `ARM_` prefix.

Comment (vpisarev) [reply cmake/OpenCVCompilerOptimizations.cmake]: fixed

**Comment by vpisarev** [comment]
@asmorkalov, @opencv-alalek, thank you for the comments and for the links! all your concerns have been addressed :)

**Pull request merged**